cristin-person/src/main/java/no/unit/nva/cristin/person/model/nva/Person.java

```diff
@@ -0,0 +1,209 @@
+            && sortedListOfIdentifiers(getIdentifiers()).equals(sortedListOfIdentifiers(that.getIdentifiers()))
+            && sortedListOfIdentifiers(getNames()).equals(sortedListOfIdentifiers(that.getNames()))
+            && Objects.equals(getContactDetails(), that.getContactDetails())
+            && Objects.equals(getImage(), that.getImage())
+            && sortedListOfAffiliations(getAffiliations()).equals(sortedListOfAffiliations(that.getAffiliations()));
+    }
+
+    @JacocoGenerated
+    @Override
+    public int hashCode() {
+        return Objects.hash(getContext(), getId(), sortedListOfIdentifiers(getIdentifiers()),
+            sortedListOfIdentifiers(getNames()), getContactDetails(), getImage(),
+            sortedListOfAffiliations(getAffiliations()));
+    }
+
+    private static URI extractImage(CristinPerson cristinPerson) {
+        return attempt(() -> new URI(cristinPerson.getPictureUrl())).orElse(uriFailure -> null);
+    }
+
+    private List<NvaIdentifier> sortedListOfIdentifiers(List<NvaIdentifier> listToSort) {
+        return listToSort.stream().sorted(Comparator.comparing(NvaIdentifier::getType)).collect(Collectors.toList());
+    }
+
+    @JacocoGenerated
+    public static final class Builder {
+
+        private transient String context;
+        private transient URI id;
+        private transient List<NvaIdentifier> identifiers;
+        private transient List<NvaIdentifier> names;
+        private transient ContactDetails contactDetails;
+        private transient URI image;
+        private transient List<Affiliation> affiliations;
+
+        public Builder withContext(String context) {
+            this.context = context;
+            return this;
+        }
+
+        public Builder withId(URI id) {
+            this.id = id;
+            return this;
+        }
+
+        public Builder withIdentifiers(List<NvaIdentifier> identifiers) {
+            this.identifiers = identifiers;
+            return this;
+        }
+
+        public Builder withNames(List<NvaIdentifier> names) {
+            this.names = names;
+            return this;
+        }
+
+        public Builder withContactDetails(ContactDetails contactDetails) {
+            this.contactDetails = contactDetails;
+            return this;
+        }
+
+        public Builder withImage(URI image) {
+            this.image = image;
+            return this;
+        }
+
+        public Builder withAffiliations(List<Affiliation> affiliations) {
+            this.affiliations = affiliations;
+            return this;
+        }
+
+        public Person build() {
+            return new Person(this.context, this.id, this.identifiers, this.names, this.contactDetails,
+                this.image, this.affiliations);
+        }
+    }
+
+    public void setContext(String context) {
+        this.context = context;
+    }
+
+    private static ContactDetails extractContactDetails(CristinPerson cristinPerson) {
```

Comment:
The methods for constructing this object should be part of CristinPerson and called in CristinPerson::toPerson